ntbs-service/Models/Validations/ValidationAttributes.cs

```diff
@@ -44,4 +45,30 @@ private bool isTruthy(object value)
             return value != null && (bool)value;
         }
     }
+
+    [AttributeUsage(AttributeTargets.Class)]
+    public class AtLeastOnePropertyAttribute : ValidationAttribute
+    {
+        private string[] PropertyList { get; set; }
+
+        public AtLeastOnePropertyAttribute(params string[] propertyList)
+        {
+            this.PropertyList  = propertyList;
+        }
+
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext) {
+            
+            PropertyInfo propertyInfo;

```

Comment:
As propertyInfo is not required out of scope of the foreach loop, think I would rather see it in there.

I also imagine the compiler will be clever enough to use any shortcuts like this it, if they're possible.